File: q2125/q2125.py
class Solution:
    def numberOfBeams(self, bank: List[str]) -> int:
        prevCount = 0
        totalBeam = 0
        for row in bank:
            currCount = 0 
            for cell in row: 
                if cell == "1": currCount += 1
            # Only update the totalBeam if the row has some devices
            if currCount != 0: 
                totalBeam += currCount * prevCount
                # We only need to track the last count of the devices in the row
                prevCount = currCount
        return totalBeam

File: q2125/q2125.cpp
class Solution
{
public:
    int numberOfBeams(vector<string> &bank)
    {
        int prevCount = 0;
        int totalBeam = 0;
        for (const string &s : bank)
        {
            int currCount = 0;
            for (const char c : s)
            {
                if (c == '1')
                    currCount++;
            }

            // Only update the totalBeam if the row has some devices
            if (currCount != 0)
            {
                totalBeam += prevCount * currCount;
                // We only need to track the last count of the devices in the row
                prevCount = currCount;
            }
        }
        return totalBeam;
    }
};